cl/opinion_page/forms.py

```diff
@@ -33,13 +33,15 @@ class CitationRedirectorForm(forms.Form):
         widget=forms.TextInput(
             attrs={"class": "form-control input-lg", "placeholder": "Volume"}
         ),
-        required=True,
+        required=False,
     )
+    # We change the place holder to allow people to continue to use the rest
+    # of the redirect API with no modifications.
```

Comment:
I tried that first.

One option, make a new form that just had full_citation in it.  Then make decisions based on `is_valid()` as to which form to use.  In addition, it adds some complication to the `reverse()` functions, making sure that the correct URL pattern is picked.

Another option is to add a full_citation field to the current form.  This would require all the fields to be `required=False`.  This gets confusing and might have other UI interactions.

Having tried three different methods of accomplishing this particular input processing, this produced the cleanest code.

I use a style of programming which I call "Railroad siding".  In this style, there is always a main line of progression.  Anything that moves us away from that main line moves us onto a siding.  The goal is to do as little modification as possible and then rejoin the main line.

Thus, this code checks to see if there is a full citation.  If there is a full citation, then it sets the reporter, volume, and page and rejoins the main line.  Nothing after that one check and set has major changes.

I would suggest that we keep my method this time.  If you don't like it, we'll decide on which method to use, and I'll code that.

Reply:
OK.